spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -258,6 +295,125 @@ protected abstract MessageProducer createConsumerEndpoint(ConsumerDestination de
 	protected void afterUnbindConsumer(ConsumerDestination destination, String group, C consumerProperties) {
 	}
 
+	/**
+	 * Build an errorChannelRecoverer that writes to a pub/sub channel for the destination.
+	 * @param destination the destination.
+	 * @param extendedConsumerProperties the properties.
+	 * @param errorMessageStrategy
+	 * @return the recoverer.
+	 */
+	protected final ErrorInfrastructure registerErrorInfrastructure(ConsumerDestination destination, String group,
+			ExtendedConsumerProperties<?> extendedConsumerProperties) {
+		ErrorMessageStrategy errorMessageStrategy = getErrorMessageStrategy();
+		ConfigurableListableBeanFactory beanFactory = getApplicationContext().getBeanFactory();
+		String errorChannelName = errorsBaseName(destination, group, extendedConsumerProperties);
+		SubscribableChannel errorChannel;
+		if (getApplicationContext().containsBean(errorChannelName)) {
+			errorChannel = getApplicationContext().getBean(errorChannelName, SubscribableChannel.class);
```

Comment:
Well, there is no guarantee that end-user always declare error channel as `PublishSubscribeChannel`.
Or is it a part of binding service as well here?